src/lib/fetch.d.ts

```diff
@@ -0,0 +1,129 @@
+/// The fetch API standard can be found at https://fetch.spec.whatwg.org/
+/// Definitions by: Ryan Graham <https://github.com/ryan-codingintrigue>, Kagami Sascha Rosylight <https://github.com/saschanaz>, Robin Van den Broeck <https://github.com/gamesmaxed>
+
+/// <reference path="lib.streams.d.ts" />
+
+interface Window {
+    fetch(url: RequestInfo, init?: RequestInit): Promise<Response>;
+}
+declare var fetch: typeof window.fetch;
+
+declare type HeadersInit = Headers | string[][] | { [key: string]: string };

```

Comment:
For autocomplete maybe we add list of [known headers](https://en.wikipedia.org/wiki/List_of_HTTP_header_fields#Request_fields)

```
type KnownHeaders = 'Accept' | 'Accept-Charset' //...
```